src/main/java/blitz/rijks/RijksSearchFrame.java

```diff
@@ -0,0 +1,147 @@
+                currentPage++;
+                searchArtworks();
+            }
+        });
+
+        search.addActionListener(new ActionListener() {
+            @Override
+            public void actionPerformed(ActionEvent e) {
+                currentPage = 1;
+                searchArtworks();
+            }
+        });
+
+        resultPanel.setLayout(new GridLayout(0, 4, 10, 10));
+        service = new RijksServiceFactory().getService();
+        searchArtworks();
+    }
+
+    private void searchArtworks() {
+        ApiKey apiKey = new ApiKey();
+        String query = search.getText().trim();
+        if (query.isEmpty()) {
+            service.getPage(apiKey.get(), currentPage)
+                    .subscribeOn(Schedulers.io())
+                    .observeOn(Schedulers.from(SwingUtilities::invokeLater))
+                    .subscribe(this::updateResults, throwable ->
+                            showError(throwable.getMessage()));
+        } else {
+            service.query(apiKey.get(), currentPage, query)
+                    .subscribeOn(Schedulers.io())
+                    .observeOn(Schedulers.from(SwingUtilities::invokeLater))
+                    .subscribe(this::updateResults, throwable ->
+                            showError(throwable.getMessage()));
+        }
+    }
+
+    private void updateResults(ArtObjects response) {
+        resultPanel.removeAll();
+        for (ArtObject artObject : response.artObjects) {
+            try {
+                URL url = new URL(artObject.webImage.url);
+                Image image = ImageIO.read(url);
+                Image scaledImage = image.getScaledInstance(
+                        200, -1, Image.SCALE_SMOOTH);
+                ImageIcon imageIcon = new ImageIcon(scaledImage);
+                JLabel imageLabel = new JLabel(imageIcon);
+                imageLabel.setText(artObject.title);
+                imageLabel.setToolTipText("<html>Title: " + artObject.title
+                        + "<br>Artist: " + artObject.principalOrFirstMaker + "</html>");
+                imageLabel.setHorizontalTextPosition(JLabel.CENTER);
+                imageLabel.setVerticalTextPosition(JLabel.BOTTOM);
+                imageLabel.addMouseListener(new java.awt.event.MouseAdapter() {
+                    public void mouseClicked(java.awt.event.MouseEvent evt) {
+                        showLargeImage(artObject);
+                    }
+                });
+                resultPanel.add(imageLabel);
+            } catch (IOException e) {
+                e.printStackTrace();
+            }
+        }
+        resultPanel.revalidate();
+        resultPanel.repaint();
+    }
+
+    private void showError(String message) {
+        JOptionPane.showMessageDialog(this,
+                "Failed to fetch data: " + message,
+                "Error", JOptionPane.ERROR_MESSAGE);
+    }
+
+    private void showLargeImage(ArtObject artObject) {
+        JFrame frame = new JFrame(artObject.title);
```

Comment:
Even though it's technically correct this is not how you were taught to make JFrames.